grappelli/templatetags/grp_tags.py

```diff
@@ -123,23 +123,28 @@ def classpath(obj):
 
 # FORMSETSORT FOR SORTABLE INLINES
 
+
 @register.filter
 def formsetsort(formset, arg):
     """
     Takes a list of formset dicts, returns that list sorted by the sortable field.
     """
+
     if arg:
         sorted_list = []
+        unsorted_list = []
         for item in formset:
-            position = item.form[arg].data
-            if position and position != "-1":
+            position = item.form[arg].value()
+
+            if isinstance(position, int) and item.original: # normal view
+                sorted_list.append((position, item))
+            elif position and item.form.cleaned_data:       # error validation view
```

Comment:
`item.form.cleaned_data` might not exist and could raise `AttributeError`. Perhaps `hasattr(item.form, 'cleaned_data')` might be better?

Reply:
i fixed that. thanks